docs/pluggable-discovery-specification.md

```diff
@@ -0,0 +1,368 @@
+or incomplete list.
+
+If an error occurs and the discovery can't complete the enumeration is must report the error with:
+
+```JSON
+{
+  "eventType": "list",
+  "error": true,
+  "message": "Resource busy"
+}
+```
+
+The `error` field must be set to `true` and the `message` field should contain a description of the error.
+
+#### START_SYNC command
+
+The `START_SYNC` command puts the tool in "events" mode: the discovery will send `add` and `remove` events each time a
+new port is detected or removed respectively. If the discovery goes into "events" mode successfully the response to this
+command is:
+
+```JSON
+{
+  "eventType": "start_sync",
+  "message": "OK"
+}
+```
+
+After this message the discoery will send `add` and `remove` event asyncronoushly (more on that later). If an error
+occurs and the discovery can't go in "events" mode the error must be reported as:
+
+```JSON
+{
+  "eventType": "start_sync",
+  "error": true,
+  "message": "Resource busy"
+}
+```
+
+The `error` field must be set to `true` and the `message` field should contain a description of the error.
+
+Once in "event" mode, the discovery is allowed to send `add` and `remove` messages asynchronously in realtime, this
+means that the client must be able to handle these incoming messages at any moment.
+
+The `add` event looks like the following:
+
+```JSON
+{
+  "eventType": "add",
+  "port": {
+    "address": "/dev/ttyACM0",
+    "label": "ttyACM0",
+    "properties": {
+      "pid": "0x804e",
+      "vid": "0x2341",
+      "serialNumber": "EBEABFD6514D32364E202020FF10181E",
+      "name": "ttyACM0"
+    },
+    "protocol": "serial",
+    "protocolLabel": "Serial Port (USB)"
+  }
+}
+```
+
+It basically provides the same information as the `list` event but for a single port. After calling `START_SYNC` an
+initial burst of add events must be generated in sequence to report all the ports available at the moment of the start.
+
+The `remove` event looks like the following:
+
+```JSON
+{
+  "eventType": "remove",
+  "port": {
+    "address": "/dev/ttyACM0",
+    "protocol": "serial"
+  }
+}
+```
+
+The content is straightforward, in this case only the `address` and `protocol` fields are reported.
+
+If the information about a port needs to be updated the discovery may send a new `add` message for the same port address
+and protocol without sending a `remove` first: this means that all the previous information about the port must be
+discarded and replaced with the new one.
+
+#### Invalid commands
+
+If the client sends an invalid or malformed command, the discovery should answer with:
+
+```JSON
+{
+  "eventType": "command_error",
+  "error": true,
+  "message": "Unknown command XXXX"
+}
+```
+
+### Board identification
+
+The `properties` associated to a port can be used to identify the board attached to that port. The algorithm is simple:
+
+- each board listed in the platform file `boards.txt` may declare a set of `upload_port.*` properties
```

Comment:
```suggestion
- each board listed in the platform file [`boards.txt`](platform-specification.md#boardstxt) may declare a set of
  `upload_port.*` properties
```
The `upload_port.*` properties need to be documented in the platform specification.

Reply:
@silvanocerza I see that my suggestion to linkify `boards.txt` was accepted, but I still didn't find any mention of the `upload_port` properties in the platform specification, so I think that has not been done.

Reply:
Yeah, completely forgot to add that on the platform specification.

Reply:
Added in https://github.com/arduino/arduino-cli/commit/efdb11c5cb7f7945e47a515f704830ec931f3fc5

Reply:
Thanks!